lib/src/easy_localization_app.dart

```diff
@@ -0,0 +1,146 @@
+  Locale get locale => _locale;
+  List<Locale> get supportedLocales => widget.supportedLocales;
+  _EasyLocalizationDelegate get delegate => widget.delegate;
+
+  @override
+  void initState() {
+    super.initState();
+    changeLocale();
+  }
+
+  // locale is either the deviceLocale or the MaterialApp widget locale.
+  // This function is responsible for returning a locale that is supported by your app
+  // if the app is opened for the first time and we only have the deviceLocale information.
+  Locale localeResolutionCallback(
+      Locale locale, Iterable<Locale> supportedLocales) {
+    if (supportedLocales != widget.supportedLocales)
+      throw new Exception(
+          "You haven't given EasyLocalization.supportedLocales to MaterialApp ssupportedLocales property");
+
+    if (supportedLocales.contains(locale)) return locale;
+
+    return widget.fallbackLocale;
+  }
+
+  void changeLocale({Locale locale}) async {
+    var _defaultLocal;
+    print(
+        "================== changeLocale ${locale?.toLanguageTag()}===================");
+    try {
+      _preferences = await SharedPreferences.getInstance();
+      if (locale == null) {
+        var _codeLang = _preferences.getString('codeLa');
+        var _codeCoun = _preferences.getString('codeCa');
+        if (_codeLang == null) {
+          var currentLocale = Intl.getCurrentLocale().split("_");
+          _defaultLocal = Locale(currentLocale[0], currentLocale[1]);
+        } else {
+          _defaultLocal = Locale(_codeLang, _codeCoun);
+        }
+      } else
+        _defaultLocal = locale;
+      await _preferences.setString('codeCa', _defaultLocal.countryCode);
+      await _preferences.setString('codeLa', _defaultLocal.languageCode);
+
+      print(_defaultLocal.toString());
+
+      setState(() {
+        _locale = _defaultLocal;
+      });
+    } catch (e) {
+      print(e);
+    }
+    print("================== changeLocale //===================");
+  }
+
+  @override
+  Widget build(BuildContext context) => _EasyLocalizationProvider(
+        data: this,
+        child: widget.child,
+      );
+}
+
+class _EasyLocalizationDelegate extends LocalizationsDelegate<Localization> {
+  final String path;
+  final String loadPath;
+  final AssetLoader assetLoader;
+
+  ///  * use only the lang code to generate i18n file path like en.json or ar.json
+  final bool useOnlyLangCode;
+
+  _EasyLocalizationDelegate({
+    this.path,
+    this.loadPath,
+    this.useOnlyLangCode = false,
+    this.assetLoader = const RootBundleAssetLoader(),
+  });
+
+  @override
+  bool isSupported(Locale locale) => locale != null;
+
+  @override
+  Future<Localization> load(Locale value) async {
+    await Localization.load(
+      value,
+      path: path,
+      useOnlyLangCode: useOnlyLangCode,
+      assetLoader: assetLoader,
+    );
+    return Localization.instance;
+  }
+
+  @override
+  bool shouldReload(LocalizationsDelegate<Localization> old) => false;
```

Comment:
This one was returning `true` previously. Why did you change that?

Reply:
I am not sure this about this one. But in the examples i have seen e.g. GlobalMaterialLocalizations false is used.
/// Returns true if the resources for this delegate should be loaded
  /// again by calling the [load] method.
  ///
  /// This method is called whenever its [Localizations] widget is
  /// rebuilt. If it returns true then dependent widgets will be rebuilt
  /// after [load] has completed.

I don't see why a LocalizationsDelegate shouldReload

Reply:
When the user forces a locale change, shouldn't load be called again and all widgets rebuild? So they automatically use the new locale